pkg/listener/listener_test.go

```diff
@@ -82,38 +94,46 @@ func Test_listlisteners(t *testing.T) {
 	assert.Assert(t, is.Contains(listenerStr, "localhost:10161"), "Expected to find device1 in list. Got %s", listeners)
 	assert.Assert(t, is.Contains(listenerStr, "localhost:10162"), "Expected to find device2 in list. Got %s", listeners)
 	assert.Assert(t, is.Contains(listenerStr, "localhost:10163"), "Expected to find device3 in list. Got %s", listeners)
+	tearDown(d)
 }
 
 func Test_register(t *testing.T) {
-	device4Channel, err := Register("device4", true)
+	d := setUp()
+	device4Channel, err := d.Register("device4", true)
 	assert.NilError(t, err, "Unexpected error when registering device %s", err)
 
 	var deviceChannelIf interface{} = device4Channel
 	chanType, ok := deviceChannelIf.(chan events.ConfigEvent)
 	assert.Assert(t, ok, "Unexpected channel type when registering device %v", chanType)
-
+	d.Unregister("device4", true)
+	tearDown(d)
 }
 
 func Test_unregister(t *testing.T) {
-	err1 := Unregister("device5", true)
+	d := setUp()
+	err1 := d.Unregister("device5", true)
 
 	assert.Assert(t, err1 != nil, "Unexpected lack of error when unregistering non existent device")
 	assert.Assert(t, is.Contains(err1.Error(), "had not been registered"), "Unexpected error text when unregistering non existent device %s", err1)
 
-	_, err1 = Register("device6", true)
```

Comment:
No need for a LHS on these register calls, since you are not checking the results anyway. Go report card flags these as "ineffectual assignments"